src/aioquic/quic/congestion/cubic.py

```diff
@@ -0,0 +1,206 @@
+        target_segments = K_CUBIC_C * (t - self.K)**3 + (W_max_segments)
+        return target_segments * self._max_datagram_size
+    
+    def is_reno_friendly(self, t) -> bool:
+        return self.W_cubic(t) < self._W_est
+    
+    def is_concave(self):
+        return self.congestion_window < self._W_max
+    
+    def reset(self):
+        self.congestion_window = K_INITIAL_WINDOW * self._max_datagram_size
+        self.ssthresh = None
+
+        self._first_slow_start = True
+        self._starting_congestion_avoidance = False
+        self.K = 0
+        self._W_est = 0
+        self._cwnd_epoch = 0
+        self._t_epoch = 0
+        self._W_max = self.congestion_window
+
+    def on_packet_acked(self, packet: QuicSentPacket) -> None:
+        self.bytes_in_flight -= packet.sent_bytes
+        self.last_ack = packet.sent_time
+        now = packet.sent_time
+
+        if self.ssthresh is None or self.congestion_window < self.ssthresh:
+            # slow start
+            self.congestion_window += packet.sent_bytes
+        else:
+            # congestion avoidance
+            if (self._first_slow_start and not self._starting_congestion_avoidance):
+                # exiting slow start without having a loss
+                self._first_slow_start = False
+                self._W_max = self.congestion_window
+                self._t_epoch = now
+                self._cwnd_epoch = self.congestion_window
+                self._W_est = self._cwnd_epoch
+                # calculate K
+                W_max_segments = self._W_max / self._max_datagram_size
+                cwnd_epoch_segments = self._cwnd_epoch / self._max_datagram_size
+                self.K = better_cube_root((W_max_segments - cwnd_epoch_segments)/K_CUBIC_C)
+
+            # initialize the variables used at start of congestion avoidance
+            if self._starting_congestion_avoidance:
+                self._starting_congestion_avoidance = False
+                self._first_slow_start = False
+                self._t_epoch = now
+                self._cwnd_epoch = self.congestion_window
+                self._W_est = self._cwnd_epoch
+                # calculate K
+                W_max_segments = self._W_max / self._max_datagram_size
+                cwnd_epoch_segments = self._cwnd_epoch / self._max_datagram_size
+                self.K = better_cube_root((W_max_segments - cwnd_epoch_segments)/K_CUBIC_C)
+
+
+            self._W_est = self._W_est + self.additive_increase_factor*(packet.sent_bytes/self.congestion_window)
+
+            t = now - self._t_epoch
+            
+            target = None
+            if (self.W_cubic(t + self.rtt) < self.congestion_window):
+                target = self.congestion_window
+            elif (self.W_cubic(t + self.rtt) > 1.5*self.congestion_window):
+                target = self.congestion_window*1.5
+            else:
+                target = self.W_cubic(t + self.rtt)
```

Comment:
No need to re-calculate this 3 times, just assign `target = self.W_cubic(t + self.rtt)` then clamp it